app/controllers/concerns/authenticable.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Authenticable
+  ENTITIES = [User].freeze
+
+  extend ActiveSupport::Concern
+
+  ENTITIES.each do |entity|
+    entity_name = entity.to_s.downcase
+
+    define_method("#current_#{entity_name}") do
+      instance_variable_get("@current_#{entity}")
```

Comment:
`entity_name`?

Reply:
Yup, I will have to add couple of specs in the future xD